homeassistant/components/lifx/util.py

```diff
@@ -80,6 +82,17 @@ def find_hsbk(hass: HomeAssistant, **kwargs: Any) -> list[float | int | None] |
     """
     hue, saturation, brightness, kelvin = [None] * 4
 
+    if (color_name := kwargs.pop(ATTR_COLOR_NAME, None)) is not None:
```

Comment:
```suggestion
    if color_name := kwargs.get(ATTR_COLOR_NAME):
```

Any reason we need to pop it?

Reply:
It's the same code as is used in `preprocess_turn_on_alternatives`. I removed the call to that method because it breaks the use of `kelvin` correctly, so I'm just reinstating the bits of the preprocessing that we still need.